pkg/cmd/provider/install.go

```diff
@@ -77,16 +78,25 @@ var providerInstallCmd = &cobra.Command{
 		}
 
 		downloadUrls := convertToStringMap((*providersManifest)[*providerToInstall.Name].Versions[*providerToInstall.Version].DownloadUrls)
-		res, err = apiClient.ProviderAPI.InstallProviderExecute(apiclient.ApiInstallProviderRequest{}.Provider(apiclient.InstallProviderRequest{
-			Name:         providerToInstall.Name,
-			DownloadUrls: &downloadUrls,
-		}))
-		if err != nil {
-			log.Fatal(apiclient_util.HandleErrorResponse(res, err))
-		}
+		err = view_utils.With(func() error {
```

Comment:
I just noticed some confusing naming in the existing spinner package and changed this function to "WithSpinner" to make it more clear. Please do a rebase from main to fix this.

One change I would like to propose is to change the text from "Loading" to "Installing". I believe an easy way to do this is to add a property to the view model and have the label passed into WithSpinner

Reply:
Added the changes pls check @idagelic.